src/ILCompiler/src/Program.cs

```diff
@@ -170,6 +188,51 @@ private int Run(string[] args)
             return 0;
         }
 
+        private TypeDesc FindType(CompilerTypeSystemContext context, string typeName)
+        {
+            TypeDesc foundType = context.SystemModule.GetTypeByCustomAttributeTypeName(typeName);
+            if (foundType == null)
+            {
+                throw new CommandLineException($"Type '{typeName}' not found");
+            }
+            return foundType;
+        }
+
+        private MethodDesc CheckAndParseSingleMethodModeArguments(CompilerTypeSystemContext context)
+        {
+            if (_singleMethodName == null && _singleMethodTypeName == null && _singleMethodGenericArgs == null)
+                return null;
+
+            if (_singleMethodName == null || _singleMethodTypeName == null)
+                throw new CommandLineException("Both method name and type name are required parameters for single method mode");
```

Comment:
It would be nice to parse the type / method name as a single string instead of a collection of pieces and if a method fails to compile, write out the name in the same parseable format a la Nutc